Sources/Client/Client/Client+Events.swift

```diff
@@ -0,0 +1,150 @@
+//
+//  Client+Events.swift
+//  StreamChatClient
+//
+//  Created by Bahadir Oncel on 30.03.2020.
+//  Copyright © 2020 Stream.io Inc. All rights reserved.
+//
+
+import Foundation
+
+/// Reference for the subscription initiated. Call `cancel()` to end subscription.
+public protocol Cancellable {
+    /// Cancel the underlying subscription for this object.
+    func cancel()
+}
+
+/// Reference for the subscription initiated. Call `cancel()` to end the subscription. Alternatively, when this object is deallocated, it calls `cancel()` on itself automatically.
+public protocol AutoCancellable: Cancellable {}
+
+struct Subscription: Cancellable {
+    private let onCancel: (String) -> ()
+    let uuid: String
+    
+    init(onCancel: @escaping (String) -> ()) {
+        self.onCancel = onCancel
+        uuid = UUID().uuidString
+    }
+    
+    public func cancel() {
+        onCancel(uuid)
+    }
+}
+
+class SubscriptionBag: Cancellable {
+    private var subscriptions = [Cancellable]()
+    
+    public func add(_ subscription: Cancellable)  {
```

Comment:
why it's public?

Reply:
I think this class should be public so people can make use of it (managing multiple subscriptions), else they'll do `[Cancellable]` themselves which doesn't make sense since we have this.

Reply:
I agree with you, but now you missed to make `public class` itself.

Reply:
Yeah I'll make it `public final class` 👍🏼